src/main/scala/clump/Clump.scala

```diff
@@ -17,9 +17,9 @@ sealed trait Clump[+T] {
 
   def join[U](other: Clump[U]): Clump[(T, U)] = new ClumpJoin(this, other)
 
-  def handle[B >: T](f: Throwable => B): Clump[B] = new ClumpHandle(this, f)
+  def handle[B >: T](f: PartialFunction[Throwable, Option[B]]): Clump[B] = new ClumpHandle(this, f)
```

Comment:
I also changed this method to allow the user to return Some or None, the previous implementation was assuming that all fallbacks would be not empty.